Docs: Updating UI extensions how-to guides with instructions on adding static information in plugin.json (#2199)
grafana/plugin-tools: docusaurus/docs/how-to-guides/ui-extensions/register-an-extension.md

sidebar_position: 20
---

> Extensions are links or React components in an app plugin. Extensions are linked to an extension point and they either render in the core Grafana UI or in another app plugin. They can also work as functions returning values.

Reviewer: ```suggestion
Extensions are links, functions or React components defined in an app plugin. Each extension is associated with an extension point and can render either in the core Grafana UI or within another app plugin.
```

## Troubleshooting

If you cannot see your link or component extension check the following:

1. **Check the console logs** - your link or component may not be appearing due to validation errors. Look for the relevant logs in your browser's console.
1. **Check the `targets`** - make sure that you are using the correct extension point IDs, and always use the `PluginExtensionPoints` enum for Grafana extension points.
1. **Check the links `configure()` function** - if your link has a `configure()` function which is returning `undefined`, the link is hidden.
1. **Check your component's implementation** - if your component returns `null` it won't be rendered at the extension point.
1. **Check if you register too many links or components** - certain extension points limit the number of links or components allowed per plugin. If your plugin registers more links or components for the same extension point than the allowed amount, some of them may be filtered out.

> 1. **Check the Grafana version** - link and component extensions are only supported after Grafana version **`>=10.1.0`**. `addLink()` and `addComponent()` are only supported in versions **>=`11.1.0`**.

Reviewer: What about functions?

Author: Great catch, I will have a look and update the docs.
